src/workerd/api/pyodide/pyodide.c++

```diff
@@ -102,6 +116,176 @@ int ArtifactBundler::readMemorySnapshot(int offset, kj::Array<kj::byte> buf) {
+      //
+      // We also accept `.` because import idents can contain it.
+      if (std::isdigit(str[start])) {
+        return 0;
+      }
+      int i = 0;
+      for (; start + i < str.size(); i++) {
+        char c = str[start + i];
+        bool validIdentChar = std::isalpha(c) || std::isdigit(c) || c == '_' || c == '.';
+        if (!validIdentChar) {
+          return i;
+        }
+      }
+
+      return i;
+    };
+
+    int i = 0;
+    while (i < file.size()) {
+      auto keywordToParse = file[i] == 'i' ? "import"_kj : "from"_kj;
+      switch (file[i]) {
+        case ' ':
+        case '\t':
+          // Indented line, skip to newline.
+          i += skipUntil(file, {'\n', '\r'}, i);
+          if (file[i] != '\0') {
+            i += skipChar(file, {'\n', '\r'}, i);  // skip newline.
+          }
+          break;
+        case '\n':
+          // Windows users may submit \r\n newlines, block above would skip the `\r` so we'll be
+          // left with `\n` which we skip explicitly here.
+          i += 1;
+          break;
+        case 'i':
+        case 'f':
+          if (!parseKeyword(file, keywordToParse, i)) {
+            i++;
+            continue;
+          }
+          i += keywordToParse.size();  // skip "import" or "from"
+
+          while (i < file.size()) {
+            // Python expects a `\` to be paired with a newline, but we don't have to be as strict
+            // here.
+            i += skipWhile(
+                file, {'\r', '\n', ' ', '\t', '\\'}, i);  // skip whitespace and backslash.
+
+            if (file[i] == '.') {
+              // ignore relative imports
+              break;
+            }
+
+            int identLen = parseIdent(file, i);
+            KJ_REQUIRE(identLen > 0);
+
+            kj::String ident = kj::heapString(file.slice(i, i + identLen));
+            if (ident[identLen - 1] != '.') {  // trailing period means the import is invalid
+              result.add(kj::mv(ident));
+            }
+
+            i += identLen;
+
+            // If "import" statement then look for comma.
+            if (keywordToParse == "import") {
+              i += skipWhile(
+                  file, {'\r', '\n', ' ', '\t', '\\'}, i);  // skip whitespace and backslash.
+              // Check if next char is a comma.
+              if (file[i] == ',') {
+                i += 1;  // Skip comma.
+                // Allow while loop to continue
+              } else {
+                // No more idents, so break out of loop.
+                break;
+              }
+            } else {
+              // The "from" statement doesn't support commas.
+              break;
+            }
+          }
+          break;
+        default:
+          // Skip to the next line.
+          i += skipUntil(file, {'\n', '\r'}, i);
+          if (file[i] != '\0') {
+            i += skipChar(file, {'\n', '\r'}, i);  // skip newline.
```

Comment:
[nit] Again I think you can get rid of this in favour of handling in `switch` statement, unless we want/need to ensure that we don't get consecutive `\r`s, for example.

Reply:
done, removed all branches except the `default`